# Normalização de dados de vendas

Este notebook limpa e normaliza o arquivo `vendas_2023_2024.csv`, preparando-o para análises posteriores.

## Tarefas
- Carregar o CSV
- Normalizar datas (formato inconsistente)
- Remover linhas com datas inválidas
- Verificar e remover duplicatas
- Salvar CSV limpo

In [1]:
import pandas as pd

## Carregar dados de vendas

In [2]:
vendas = pd.read_csv('datasets/vendas_2023_2024.csv')
print(f"Linhas originais: {len(vendas)}")
print(vendas.head())

Linhas originais: 9895
   id  id_client  id_product  qtd     total   sale_date
0   0         42         105   11    3405.0  2023-09-10
1   1          3         136    9   16873.9  15-09-2024
2   2         25         139    7    9475.3  2024-08-13
3   4         20          23    5   55893.0  2023-02-03
4   5          8          57    4  451403.9  2024-02-12


## Normalizar datas

In [ ]:
    # Função para analisar datas com múltiplos formatos
    def parse_date(date_str):
        # Tentar formato brasileiro (dd-mm-yyyy) primeiro
        try:
            return pd.to_datetime(date_str, format='%d-%m-%Y', errors='coerce')
        except:
            pass
        # Tentar formato ISO (yyyy-mm-dd)
        try:
            return pd.to_datetime(date_str, format='%Y-%m-%d', errors='coerce')
        except:
            pass
        # Fallback: deixar pandas tentar inferir
        return pd.to_datetime(date_str, errors='coerce')
    
    vendas['sale_date'] = vendas['sale_date'].apply(parse_date)
    linhas_invalidas = vendas['sale_date'].isna().sum()
    print(f"Datas inválidas encontradas: {linhas_invalidas}")
    vendas = vendas.dropna(subset=['sale_date'])
    print(f"Linhas após remoção de datas inválidas: {len(vendas)}")
    print(vendas.head())

Datas inválidas encontradas: 0
Linhas após remoção de datas inválidas: 1950
    id  id_client  id_product  qtd      total  sale_date
0    0         42         105   11     3405.0 2023-10-09
3    4         20          23    5    55893.0 2023-03-02
4    5          8          57    4   451403.9 2024-12-02
9   11         39         128    5    23254.0 2023-07-05
15  17         43          91   11  1409558.7 2023-05-09


## Verificar e remover duplicatas

In [7]:
duplicatas = vendas.duplicated().sum()
print(f"Duplicatas encontradas: {duplicatas}")
vendas = vendas.drop_duplicates()
print(f"Linhas após remoção de duplicatas: {len(vendas)}")

Duplicatas encontradas: 0
Linhas após remoção de duplicatas: 1950


## Salvar CSV limpo

In [8]:
vendas.to_csv('datasets/vendas_normalizado.csv', index=False)
print("Arquivo salvo: datasets/vendas_normalizado.csv")

Arquivo salvo: datasets/vendas_normalizado.csv


## Resumo

- Linhas originais: 9901 (9895 no arquivo em Python pois o Pandas encontrou problemas de formatação)
- Datas inválidas removidas: 0
- Duplicatas removidas: 0
- Linhas finais: 1950

Arquivo limpo salvo como `vendas_normalizado.csv`.